LWCE_Core/Src/XComLongWarCommunityEdition/Classes/LWCE_XGAbility_Extensions.uc

```diff
@@ -1566,16 +1591,43 @@ static simulated function GetShotSummary(XGAbility_Targeted kSelf, out TShotResu
 
     kResult.bKillshot = (kTarget.GetUnitHP() + kResult.iPossibleDamage) <= 0;
 
-    if (kTarget.m_bVIP) // Probably right, needs confirmation
+    if (kTarget.m_kCharacter.m_kChar.iType == eChar_Civilian)
     {
-        if (kTarget.m_kCharacter.m_kChar.iType == eChar_Civilian || kShooter.m_kCharacter.m_kChar.aProperties[eCP_MeleeOnly] != 0)
+        switch (`LWCE_TACCFG(eCivilianHitChanceCalcStyle))
         {
-            kInfo.arrHitBonusStrings.AddItem(kSelf.m_strBonusAim);
-            kInfo.arrHitBonusValues.AddItem(100);
-            return;
+            case eHCCS_AlwaysHit:
+                kInfo.arrHitBonusStrings.AddItem(kSelf.m_strBonusAim);
+                kInfo.arrHitBonusValues.AddItem(1000);
+                return;
+            case eHCCS_NeverHit:
+                kInfo.arrHitPenaltyStrings.AddItem(kSelf.m_strBonusAim);
+                kInfo.arrHitPenaltyValues.AddItem(-1000);
+                return;
+            case eHCCS_CivilianBaseGame:
```

Comment:
micro-optimization might be to reorder these to put the common basegame case first, I don't know if e.g. AI calls this frequently to evaluate its moves.